drake/examples/simple_continuous_time_system.cc

```diff
@@ -0,0 +1,81 @@
+
+/// Simple Continuous Time System Example
+///
+/// This is meant to be a sort of "hello world" example for the
+/// drake::system classes.  It defines a very simple continous time system,
+/// simulates it from a given initial condition, and plots the result.
+
+#include "drake/systems/analysis/simulator.h"
+#include "drake/systems/framework/leaf_system.h"
+
+namespace drake {
+namespace examples {
+
+/// Simple Continuous Time System
+/// \begin{gather*}
+///     \dot{x} = -x + x^3 \\
+///     y = x
+/// \end{gather*}
+class SimpleContinuousTimeSystem : public systems::LeafSystem<double> {
+ public:
+  SimpleContinuousTimeSystem() {
+    this->DeclareOutputPort(systems::kVectorValued,
+                            1,  // dimension of output (y) = 1
+                            systems::kContinuousSampling);
+  };
+  ~SimpleContinuousTimeSystem() override{};
+
+  // xdot = -x + x^3
+  void EvalTimeDerivatives(
+      const systems::Context<double>& context,
+      systems::ContinuousState<double>* derivatives) const override {
+    double x = context.get_continuous_state()->get_state().GetAtIndex(0);
+    double xdot = -x + std::pow(x, 3.0);
+    derivatives->get_mutable_state()->SetAtIndex(0, xdot);
+  }
+
+  /// y = x
+  void EvalOutput(const systems::Context<double>& context,
+                  systems::SystemOutput<double>* output) const override {
+    double x = context.get_continuous_state()->get_state().GetAtIndex(0);
+    output->GetMutableVectorData(0)->SetAtIndex(0, x);
+  }
+
+ protected:
+  // allocate a basic vector of dimension 1
+  virtual std::unique_ptr<systems::ContinuousState<double>>
+  AllocateContinuousState() const override {
```

Comment:
This method is probably the hardest for a newbie to appreciate.  And it's quite a different workflow than the way that the input/output ports are declared.  Perhaps we could have a class that provides a default implementation of this, and the number of states is declared (e.g. in the constructor)?